01_fib/harirakr/main_test.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNegativeFib(t *testing.T) {
+	assert.Equal(t, []int64{1, -1, 2, -3, 5, -8, 13, -21, 34, -55}, fib(-10))
+	assert.Equal(t, []int64{1, -1, 2}, fib(-3))
+	assert.Equal(t, []int64{1, -1}, fib(-2))
+	assert.Equal(t, []int64{1}, fib(-1))
+	assert.Equal(t, []int64{}, fib(0))
+}
+
+func TestFib(t *testing.T) {
+	assert.Equal(t,
+		[]int64{1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765},
+		fib(20))
+	assert.Equal(t, []int64{1, 1, 2, 3, 5, 8, 13, 21, 34, 55}, fib(10))
+	assert.Equal(t, []int64{1, 1}, fib(2))
+	assert.Equal(t, []int64{1}, fib(1))
+	assert.Equal(t, []int64{}, fib(0))
+}
+
+func TestMain(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := strconv.Quote("[1 1 2 3 5 8 13]\n")
+	actual := strconv.Quote(buf.String())
+	if expected != actual {
+		t.Errorf("main fail; expected=%v, actual=%v", expected, actual)
+	}
+}
+
+// Test input out of bounds
+func TestFibOutOfBounds(t *testing.T) {
```

Comment:
The code here is duplicated.
I'd prefer to have a loop inside this test.

sth like:
```
for _, val := range []int{-94,-93,93,93} {
        var buf bytes.Buffer
	out = &buf
	fib(val)
	expected := strconv.Quote("fib: Range out of bounds; takes input > -93 and < 93\n")
	actual := strconv.Quote(buf1.String())
	if expected != actual {
		t.Errorf("Test fail fib(%d). Expected=%v, Actual=%v", val, expected, actual)
	}
}
```
or even more fancy:
```
for _, val := range []int{-94,-93,93,93} {
  t.Run(fmt.Sprintf("val%s", val), func(t *testing.T) {
        var buf bytes.Buffer
	out = &buf
	fib(val)
	expected := strconv.Quote("fib: Range out of bounds; takes input > -93 and < 93\n")
	actual := strconv.Quote(buf1.String())
	if expected != actual {
		t.Errorf("Test fail fib(%d). Expected=%v, Actual=%v", val, expected, actual)
	}
  })
}
```

Reply:
and  `	if expected != actual {` -> assert.Equal

Reply:
Don. 
Thanks @n0npax